autoload/gopher/system.vim

```diff
@@ -5,6 +5,15 @@ let s:gotools = s:root . '/tools'         " Our Go tools.
 let s:gobin   = s:gotools . '/bin'
 let s:jobs    = []                        " List of running jobs.
 
+" Use either the '<plugin>/tools' directory to install things or use 
+" $GOBIN. The default is to place things in <plugin>/tools.
+let s:use_gotools = !get(g:, 'gopher_local_install', 0)
+
+" By default autodownload of tools happens. However, if
+" we want a 'local' install we can no longer have 
+" autodownload.
+let s:tool_autodownload = s:use_gotools is 1 ? 1 : 0
```

Comment:
Did you run in to a problem with the autodownload and the default GOBIN, or is it just something you'd prefer not to happen?

I think this could perhaps be two different settings: "use default GOBIN" and "don't download tools automatically"?

Reply:
This is all true.  The amount of refactoring that would be required to do that seemed greater than I was willing to devote when I was working on this PR.  This PR was primarily a "here is an idea I have, will you accept it?"  If you are okay with the concept behind the PR I can work on it. Or you or anyone else can. I think allowing auto-download while also placing tools in default $GOBIN sounds fine.  The problem you run in to is that default $GOBIN is not a single path, but can be a list of paths (or no path).  So determining where exactly go would download tools to can be difficult.  Unless there is a way to programatically determine this. I tried

`go env GOBIN`

but this doesn't return the default value if $GOBIN is unset.

Perhaps the solution is more obvious than I think it is.

Reply:
Thinking about the $GOPATH issue sent me down a rabbit hole. This seems to be the best answer:
https://github.com/golang/go/issues/23439
Hopefully go 1.15 will make it easier to determine where tools are installed when $GOBIN is unset.

Reply:
So I mentioned all of the above because s:tool() performs an existence check on the tool using filereadable(). It requires the full path to the tool. This can be re-written using something like exepath() and there is no longer a need for the path at all.